terraform/eval_output.go

```diff
@@ -76,16 +79,20 @@ func (n *EvalWriteOutput) Eval(ctx EvalContext) (interface{}, error) {
 		}
 	}
 
-	// If it is a list of values, get the first one
-	if list, ok := valueRaw.([]interface{}); ok {
-		valueRaw = list[0]
-	}
-	if _, ok := valueRaw.(string); !ok {
-		return nil, fmt.Errorf("output %s is not a string", n.Name)
+	switch valueTyped := valueRaw.(type) {
+	case string:
+		mod.Outputs[n.Name] = valueTyped
+	case []interface{}:
+		mod.Outputs[n.Name] = valueTyped[0]
```

Comment:
This is still the case for all the interpolation functions that still return StringList right now - it can _probably_ be removed once they are converted over to the new way of doing things.